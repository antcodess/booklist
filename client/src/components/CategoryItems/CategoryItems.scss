.category-items-container {
  margin-top: 20px;
  
  h2 {
    text-align: left;
    margin-bottom: 40px; 
    font-size: 2rem;
  }
  
  .category-items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 10px;
    
    .directory-item {
      margin-bottom: 20px;
    }
  }
} 