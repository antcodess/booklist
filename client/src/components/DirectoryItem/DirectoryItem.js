import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cart.actions';

import CustomButton from '../CustomButton/CustomButton';

import './DirectoryItem.scss';

const DirectoryItem = ({ item, history, addItem }) => {
  const { title, imageUrl, price } = item;
  return (
    <div className='directory-item'>
      <div className='directory-item-image' onClick={() => history.push(`/${item.routeName}`)}>
        <img className='image' src={ imageUrl } alt='products' />
      </div>
      <div className='directory-meta' onClick={() => history.push(`/${item.routeName}`)}>
        <h4>{ title }</h4>
        <span>${ price }</span>
      </div>
      <CustomButton onClick={() => addItem(item)}>Add To Cart</CustomButton>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(null, mapDispatchToProps)(withRouter(DirectoryItem));