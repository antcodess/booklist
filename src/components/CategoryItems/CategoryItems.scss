.category-items-container {
  margin-top: 20px;
  
  h2 {
    text-align: left;
    margin-bottom: 40px; 
  }
  .category-items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 10px;
    
    .directory-item {
      margin-bottom: 20px;
    }
  }
} 