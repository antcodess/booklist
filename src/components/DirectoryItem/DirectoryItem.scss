.directory-item {
  position: relative;
  cursor: pointer;

  .directory-item-image {

    .image {
      border-radius: 10px;
      max-width: 100%;
      height: auto;
    }
  }

  h4 {
    margin-top: 10px;
    margin-bottom: 2px;
  }
  
  .custom-button {
    display: none;
    width: 90%;
    top: 58%;
    right: 5%;
    position: absolute;
  }

  &:hover {
    .custom-button {
      display: flex;
    }
    
    .image {
      opacity: 0.8;
    }
  }
}