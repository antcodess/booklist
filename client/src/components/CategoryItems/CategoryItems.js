import React from 'react';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

import './CategoryItems.scss'; 

export const CategoryItems = ({ categoryItems, categoryTitle }) => {

  return (
    <React.Fragment>
      <BreadCrumbNav title={categoryTitle} />
      <section className='category-items-container'>
        <h2>{ categoryTitle }</h2>
        <div className='category-items'>
          {categoryItems ? 
            categoryItems.map(item => <DirectoryItem key={item.id} item={item} />)
            : null }
        </div>
      </section>
    </React.Fragment> 
  )
};

export default CategoryItems;