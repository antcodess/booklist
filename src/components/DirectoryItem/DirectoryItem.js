import React from 'react';


import './DirectoryItem.scss';

const DirectoryItem = ({ photo, name, price }) => {
  return (
    <div className='directory-item'>
      <div className='directory-item-image'>
        <img src={ photo } alt='products' />
      </div>
      <h4>{ name }</h4>
      <span>{ price }</span>
    </div>
  );
};

export default DirectoryItem;