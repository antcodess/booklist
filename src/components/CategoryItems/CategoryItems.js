import React from 'react';
import { connect } from 'react-redux';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

import { titleCase } from './CategoryItems.utils';
import { selectCategoryItems } from '../../redux/shop/shop.selectors';

import './CategoryItems.scss'; 

const CategoryItems = ({ match, categoryItemsList }) => {
  const categoryItems = categoryItemsList[0];
  const categoryTitle = titleCase(decodeURIComponent(match.params.categoryTitle).replace(/[-]/g, ' '));

  return (
    <React.Fragment>
      <BreadCrumbNav title={categoryTitle} />
      <section className='category-items-container'>
        <h2>{ categoryTitle }</h2>
        <div className='category-items'>
          {categoryItems ? 
            categoryItems.map(item => <DirectoryItem key={item.id} item={item} />)
            : null }
        </div>
      </section>
    </React.Fragment>
  )
};

const mapStateToProps = (state, ownProps) => ({
  categoryItemsList: selectCategoryItems(decodeURIComponent(ownProps.match.params.categoryTitle).replace(/[-]/g, ' '))(state)
});

export default connect(mapStateToProps)(CategoryItems);