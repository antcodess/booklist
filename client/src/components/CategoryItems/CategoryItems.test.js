import React from 'react';
import { shallow } from 'enzyme';
import { CategoryItems } from './CategoryItems';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

describe('CategoryItems Component', () => {
  let wrapper;
  let mockCategoryItems;
  let mockCategoryTitle;
  
  beforeEach(() => {

    mockCategoryItems = [{id: 1}, {id: 2}, {id: 3}];
    mockCategoryTitle = 'Category';

    const mockProps = {
      categoryItems: mockCategoryItems, 
      categoryTitle: mockCategoryTitle
    };

    wrapper = shallow(<CategoryItems { ...mockProps } />);
  });

  it('should render CategoryItems Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Directory Item if component has categoryItems', () => {
    expect(wrapper.exists(DirectoryItem)).toBe(true);
  });
  
  it('should not render a Directory Item if categoryItems are empty', () => {
    const mockProps = {
      categoryItems: undefined, 
      categoryTitle: mockCategoryTitle
    };
    wrapper = shallow(<CategoryItems { ...mockProps } />)
    expect(wrapper.exists(DirectoryItem)).toBe(false);
    expect(wrapper.find('.category-items').children().length).toEqual(0);
    expect(mockProps.categoryItems).toBeUndefined();
  });
});