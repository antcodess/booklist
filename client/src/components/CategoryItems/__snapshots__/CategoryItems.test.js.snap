// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItems Component should render CategoryItems Component 1`] = `ShallowWrapper {}`;
